# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1463770395 25200
#      Fri May 20 11:53:15 2016 -0700
# Branch stable
# Node ID 212783dc8c9a7c5c666d6fdafb67edccadd5dc92
# Parent  3c09cff7fa06bcf477588603229d119344adcef6
Fix popen2 error on Windows when child writes to stderr (bug #43036)

* lo-sysdep.cc (octave::sys::popen2): Ensure child inherits handle to the
current stderr stream.

diff --git a/liboctave/system/lo-sysdep.cc b/liboctave/system/lo-sysdep.cc
--- a/liboctave/system/lo-sysdep.cc
+++ b/liboctave/system/lo-sysdep.cc
@@ -122,16 +122,17 @@ octave_popen2 (const std::string& cmd, c
     }
   fildes[1] = _open_osfhandle (reinterpret_cast<intptr_t> (parentRead),
                                _O_RDONLY | _O_BINARY);
   fildes[0] = _open_osfhandle (reinterpret_cast<intptr_t> (parentWrite),
                                _O_WRONLY | _O_BINARY);
   si.dwFlags |= STARTF_USESTDHANDLES;
   si.hStdInput = childRead;
   si.hStdOutput = childWrite;
+  si.hStdError = GetStdHandle (STD_ERROR_HANDLE);
 
   // Ignore first arg as it is the command
   for (int k=1; k<args.length (); k++)
     command += " \"" + args[k] + "\"";
   OCTAVE_LOCAL_BUFFER (char, c_command, command.length () + 1);
   strcpy (c_command, command.c_str ());
   if (! CreateProcess (0, c_command, 0, 0, TRUE, 0, 0, 0, &si, &pi))
     {
